open-api/rest-catalog-open-api.yaml

```diff
@@ -2800,6 +3053,59 @@ components:
           additionalProperties:
             type: string
 
+    PlanContext:
+      type: object
+      required:
+        - select
+        - options
+      properties:
+        select:
+          description: A list of the selected columns
+          type: array
+          items:
+            type: string
+        project:
+          description: A list of the projections
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        options:
+          type: object
+          additionalProperties:
+            type: string
+        use-snapshot:
+          type: integer
+          format: int64
+        from-snapshot-id:
+          type: integer
+          format: int64
+        to-snapshot-id:
+          type: integer
+          format: int64
+        as-of-time:
+          type: integer
+          format: int64
+        case-sensitive:
+          type: boolean
+        use-ref:
+          type: string
+
+    PreplanTableRequest:
+      type: object
+      properties:
+        allOf:
```

Comment:
I would not include the `stats-fields` list in this request. That option only applies when producing scan tasks. Given I think that I would not use `PlanContext` given that this is a subset of the arguments and the addition of `plan-task` below changes whether other arguments like `snapshot-id` are allowed.

Reply:
Will fix this